libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1750,6 +1763,16 @@ IotMqttError_t IotMqtt_PublishAsync( IotMqttConnection_t mqttConnection,
             status = IOT_MQTT_STATUS_PENDING;
         }
 
+        if( ( pPublishInfo->qos == IOT_MQTT_QOS_0 ) && ( pOperation != NULL ) )
+        {
+            if( ( ( flags & MQTT_INTERNAL_FLAG_BLOCK_ON_SEND )
+                  == MQTT_INTERNAL_FLAG_BLOCK_ON_SEND ) &&
+                ( pOperation->u.operation.status == IOT_MQTT_NETWORK_ERROR ) )
+            {
+                status = IOT_MQTT_NETWORK_ERROR;
+            }
```

Comment:
In what case, the execution would reach here? Isn't it always be going to _publishQos0Sync as per line 1864.

Reply:
Yes, will remove this logic as mentioned in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1009#discussion_r445779256